casper/src/main/rholang/MakePoS.rho

```diff
@@ -183,6 +186,48 @@ in {
           }
         } |
       
+        //TODO: Fill in this contract (RHOL-462)
+        contract this(@"distribute", @purse) = {
+          Nil
+        } |
+      
+        paymentStore!((Nil, Nil, Nil, Nil)) |
+      
+        //Make it easy to query the last payment so that validators
+        //can check how much phlo there is for the deploy.
+        contract this(@"lastPayment", return) = {
+          for(@payment <- paymentStore) {
+            paymentStore!(payment) |
+            return!(payment)
+          }
+        } |
+      
+        //This is the contract that deploy payment code ultimately needs
+        //to call in order to pay for the deploy.
+        contract this(@"pay", @purse, return) = {
+          new payPurseCh, p(`rho:deploy:params`), paramsCh in {
+            clonePurse!(purse, *payPurseCh) |
+            p!(*paramsCh) |
+            for(
+              @cloneResult <- payPurseCh;
+              @codeHash, @phloRate, @userId, @timestamp <- paramsCh
+            ) {
+              match cloneResult {
+                ("Right", (payPurse, payAmount)) => {
```

Comment:
Do I understand it correctly - we use all REV-s stored in the purse time the phloRate as the phlo limit?

Reply:
Thanks for this explanation. I was not familiar with the properties of the purses. This explain a lot.